sound/soc/sof/ipc.c

```diff
@@ -787,12 +787,15 @@ struct snd_sof_ipc *snd_sof_ipc_init(struct snd_sof_dev *sdev)
 
 	/* pre-allocate message data */
 	for (i = 0; i < IPC_EMPTY_LIST_SIZE; i++) {
-		msg->msg_data = devm_kzalloc(sdev->dev, PAGE_SIZE, GFP_KERNEL);
+		msg->msg_data = (void *)
+			devm_get_free_pages(sdev->dev,
```

Comment:
@plbossart @libinyang if I read Andy's feedback correctly, i think he was hinting at doing this "at once" meaning that use devm_get_free_pages() of size IPC_EMPTY_LIST_SIZE for msg_data and reply_data and figure out which page each  one point to. or am i grossly mistaken?

Reply:
Can you guys ask for precisions? the comment could also be construed as getting multiple pages in one shot instead of in a loop, then assigning the pointers in a loop.